src/amber/controller/render.cr

```diff
@@ -11,27 +11,27 @@ module Amber::Controller
     end
 
     macro render(template = nil, partial = nil, layout = true, path = "src/views", folder = __FILE__)
-      {% if template || partial %}
-        {{filename = template || partial}}
+      {% if !(template || partial) %}
+        raise "Template or partial required!"
+      {% end %}
 
-        {% if filename.id.split("/").size > 1 %}
-          %content = render_template("#{{{filename}}}", {{path}})
-        {% else %}
-          {% if folder.id.ends_with?(".ecr") %}
-            %content = render_template("#{{{folder.split("/")[-2]}}}/#{{{filename}}}", {{path}})
-          {% else %}
-            %content = render_template("#{{{folder.split("/").last.gsub(/\_controller\.cr|\.cr/, "")}}}/#{{{filename}}}", {{path}})
-          {% end %}
-        {% end %}
+      {{filename = template || partial}}
 
-        {% if layout && !partial %}
-          content = %content
-          %content = render_template("layouts/#{{{layout.class_name == "StringLiteral" ? layout : LAYOUT}}}", {{path}})
+      {% if filename.id.split("/").size > 1 %}
+        %content = render_template("#{{{filename}}}", {{path}})
+      {% else %}
+        {% if folder.id.ends_with?(".ecr") %}
```

Comment:
what about `slang`, `mustache`, `temel` and `crikey`?

Reply:
Slang already works. We paired on this line a few months ago, in order to fix an issue with ecr working differently than lang. As far as temel, crikey, hopefully it will work with kilt unless they don't something weird at which point we might have to handle them specifically too.

Reply:
oh yes, I recall.  That looks so out of place. oh well.